modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/ExportedTypesModelValidator.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.introspection.validation;
+
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.metadata.api.model.ObjectFieldType;
+import org.mule.metadata.api.model.ObjectType;
+import org.mule.metadata.api.visitor.MetadataTypeVisitor;
+import org.mule.runtime.api.meta.model.ComponentModel;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.api.meta.model.operation.HasOperationModels;
+import org.mule.runtime.api.meta.model.operation.OperationModel;
+import org.mule.runtime.api.meta.model.source.HasSourceModels;
+import org.mule.runtime.api.meta.model.source.SourceModel;
+import org.mule.runtime.api.meta.model.util.ExtensionWalker;
+import org.mule.runtime.extension.api.exception.IllegalModelDefinitionException;
+import org.mule.runtime.extension.api.introspection.declaration.type.TypeUtils;
+import org.mule.runtime.module.extension.internal.exception.IllegalParameterModelDefinitionException;
+import org.mule.runtime.module.extension.internal.util.IntrospectionUtils;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toSet;
+import static org.mule.metadata.api.model.MetadataFormat.JAVA;
+import static org.mule.metadata.java.api.utils.JavaTypeUtils.getType;
+import static org.mule.runtime.module.extension.internal.util.IntrospectionUtils.getFieldsWithGetters;
+
+/**
+ * Validates that the exported types used as:
+ *
+ * <ul>
+ * <li>Return type of an {@link OperationModel}</li>
+ * <li>Parameter of an {@link OperationModel}</li>
+ * <li>Return type of an {@link SourceModel}</li>
+ * </ul>
+ *
+ * These exported types should comply that for each parameter they contain, there must exist a getter method for that parameter.
+ * The viceversa is also required, that implies the fact that for each getter method, there should exist a field named with the
+ * getter convention.
+ *
+ * @since 4.0
+ */
+public final class ExportedTypesModelValidator implements ModelValidator {
+
+  @Override
+  public void validate(ExtensionModel extensionModel) throws IllegalModelDefinitionException {
+    new ExtensionWalker() {
+
+      @Override
+      public void onOperation(HasOperationModels owner, OperationModel model) {
+        model.getParameterModels().stream()
+            .forEach(parameterModel -> validateJavaType(parameterModel
+                .getType(), () -> validateParameterFieldsHaveGetters(model, parameterModel.getType())));
```

Comment:
in your original request, the validation logic was not always the same. Why is it that now you always validate with the same function? If the function is in fact always the same, then the lambda expression is overkill